gdshelpers/parts/waveguide.py

```diff
@@ -418,7 +418,10 @@ def add_route_single_circle_to(self, final_coordinates, final_angle, max_bend_st
         self._current_port = tmp_wg.current_port
 
         if not on_line_only:
-            self.add_straight_segment(distance[1] - d)
+            if final_width is not None:
+                self.add_straight_segment(length=distance[1] - d, final_width=final_width)
+            else:
+                self.add_straight_segment(distance[1] - d)
```

Comment:
This way it would only work if on_line_only is false. Therefore, I'd guess it's not the right place (only) here to add it.
Besides that, the tapering would only occur in the last segment, my expectation would be, that the tapering starts at the current_port until the given port.

Reply:
I changed the way of how the tapering is applied in the way you suggested. It now affects the complete line.